### ISSUE TITLE
fix: Report bucket/location when relevant with object store errors

### ISSUE THREAD

**Issue by carols10cents** [influxdata/influxdb_iox#140]
See https://github.com/influxdata/delorean/pull/134/files#r438680030 for discussion. `delorean_object_store` `InternalError`s should contain bucket and location associated with the error to assist in debugging.

**Comment by brandonsov** [comment]
Hello, I'm hoping to contribute to this project and take this on as a first issue. I have a few clarifying questions to determine if I understand this issue correctly before I fill out a CLA.
1. There does not exist a `delorean_object_store` in this repo on the `main` branch. To confirm, was it renamed to the `object_store` crate?
2. For modifying the S3 and Google Cloud Storage object store errors, would this involve adding new fields to the S3 related structs in the `InternalError` enum? 
3. Assuming #2 is correct, does this issue cover modifying the display of these new fields in error logging?
4. What are coverage requirements for this project in general and to this issue's code changes?

**Comment by domodwyer** [comment]
Hi @brandonsov!

You're correct `delorean_object_store` became `object_store` fairly recently :)

I just had a quick look at the errors and I think your proposal sounds good - adding the bucket/path fields to the errors sounds sensible to me, and modifying the error display to include them would be great.

As for code coverage any changes should be covered with tests - this can be tricky to do in practice, but there's a strong preference to have tests wherever possible. If it's proving tricky to test, open a WIP PR and we can strategise :) We discussed coverage in our team meeting this afternoon (or morning, depending where you are!) and we've created #531 to add code coverage checks to this repo to help track & enforce this.

We're excited to have you help! Thanks very much 🎉 

Dom
